src/elements/Label/Label.js

```diff
@@ -0,0 +1,168 @@
+    remove: (removable || onClickRemove) && <Icon className='delete' onClick={handleClickRemove} />,
+  })
+
+  // Do not destructure createElement import
+  // react-docgen only recognizes a stateless component when React.createElement is returned
+  return React.createElement(_component, _props, _children)
+}
+
+Label._meta = {
+  library: META.library.semanticUI,
+  name: 'Label',
+  type: META.type.element,
+  props: {
+    attached: ['top', 'bottom', 'top right', 'top left', 'bottom left', 'bottom right'],
+    size: sui.sizes,
+    color: sui.colors,
+    pointing: ['bottom', 'left', 'right'],
+    corner: ['left', 'right'],
+    right: ['left', 'right'],
+    ribbon: ['right'],
+  },
+}
+
+Label.propTypes = {
+  /** Attach to a <Segment />. */
+  attached: PropTypes.oneOf(Label._meta.props.attached),
+
+  /** Primary content of the label, same as text. */
+  children: PropTypes.node,
+
+  /** Classes to add to the label className. */
+  className: PropTypes.string,
+
+  /** Color of the label. */
+  color: PropTypes.oneOf(Label._meta.props.colors),
+
+  /** Place the label in one of the upper corners . */
+  corner: PropTypes.oneOf(Label._meta.props.corner),
+
+  /** Additional text with less emphasis. */
+  detail: PropTypes.string,
+
+  /** Same as detail but formatted as a link. */
+  detailLink: PropTypes.string,
+
+  /** Format a label to align better alongside text. */
+  horizontal: PropTypes.bool,
+
+  /** Float above another element in the upper right corner. */
+  floating: PropTypes.bool,
+
+  /** Make the label circular, or a dot if it is empty. */
+  circular: PropTypes.bool,
+
+  /** Add an icon by icon className or pass an <Icon /.>*/
+  icon: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.element,
+  ]),
+
+  /** Add an image by img src or pass an <Image />. */
+  image: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.element,
+  ]),
+
+  /** Format the label as a link on hover. */
+  link: PropTypes.bool,
+
+  /** Adds the link style when present, called with (event, props). */
+  onClick: PropTypes.func,
+
+  /** Click callback for detail, called with (event, props). */
+  onClickDetail: PropTypes.func,
+
+  /** Adds an "x" icon, called with (event, props) when "x" is clicked. */
+  onClickRemove: PropTypes.func,
+
+  /** Add an "x" icon that calls onClickRemove when clicked. */
+  removable: PropTypes.bool,
+
+  /** Point to content next to it. */
+  pointing: PropTypes.oneOfType([
+    PropTypes.bool,
+    PropTypes.oneOf(Label._meta.props.pointing),
+  ]),
+
+  /** Format the label as a ribbon on another component. */
+  ribbon: PropTypes.oneOf(Label._meta.props.ribbon),
+
+  /** Size of the label. */
+  size: PropTypes.oneOf(Label._meta.props.sizes),
+
+  /** Format the label like a product tag. */
+  tag: PropTypes.bool,
+
+  /** Primary text of the label, same as children. */
+  text: PropTypes.node,
+}
```

Comment:
Nope, it is actually more LOC and limits us from using life cycle methods.  However, they will take advantage of extra optimization in future versions of React.  So, we're getting on the bandwagon now.